### DISCUSSION TITLE
Can Ocelot serve up Client Credentials?

### DISCUSSION THREAD

**Discussion by PostImpatica** [ThreeMammals/Ocelot#1895] [Q&A] [answered]
Is there some way to generate and serve up client credentials from the Ocelot itself, or do I have to use an external identity provider for this?  When I ask if there is a way, I don't mean theoretically, what I mean is, is there an easy way to do this?  If so please give me a hint to get me started.  Thanks!!!

**Comment by raman-m** [comment]
@PostImpatica 
Hi Clay!
Are you still interested in this adventure?

**Reply by raman-m** [reply, marked as answer]
> is there an easy way to do this?

No! This development challenge will not be easy.
Historically Ocelot recommends external identity management tools like Identify Server.
Another options are Okta and OAuth.
In theory you can have absolutely different, any kind of Identity management as it is possible for each ASP.NET Core app, including Basic, User type, SSO services, and all types of identity management of cloud providers like Azure, Google, AWS.
We are open to other identity providers to integrate as "easy way" auth features.

Regarding client credentials... You cannot substitute or override, but we have [claims transformations](https://ocelot.readthedocs.io/en/latest/features/claimstransformation.html) feature. Look at it in docs please!
